massa-wallet/src/lib.rs

```diff
@@ -24,27 +28,39 @@ pub struct Wallet {
     pub keys: Map<Address, (PublicKey, PrivateKey)>,
     /// Path to the file containing the private keys (not encrypted)
     pub wallet_path: PathBuf,
+    /// Password
+    pub password: String,
 }
 
 impl Wallet {
     /// Generates a new wallet initialized with the provided file content
-    pub fn new(path: PathBuf) -> Result<Wallet, WalletError> {
-        let keys = if path.is_file() {
-            serde_json::from_str::<Vec<PrivateKey>>(&std::fs::read_to_string(&path)?)?
+    pub fn new(path: PathBuf, password: String) -> Result<Wallet, WalletError> {
+        let content = &std::fs::read(&path)?[..];
+        let priv_keys = if !content.is_empty() {
```

Comment:
content.is_empty() is not a good check for file existence

Reply:
this is not for file existence but only to see if the file is empty or not

Reply:
Also this is linked to https://github.com/massalabs/massa/pull/2710#discussion_r902292731, currently changing that.